File: Linkedlist/java/sumList.java
class SumList {
  public static int sumList(ListNode head) {
    int total = 0;
    ListNode current = head;
    while (current != null) {
      total += current.val;
      current = current.next;
    }
    return total;
  }

  public static int sumListRecursive(ListNode head) {
    if (head == null) {
      return 0;
    }
    return head.val + sumListRecursive(head.next);
  }

  public static void run() {
    // intentionally empty
  }
}

File: Linkedlist/csharp/SumList.cs
namespace LinkedListSolutions;

public static class SumList
{
    public static int Solve(ListNode? head)
    {
        var total = 0;
        var current = head;
        while (current is not null)
        {
            total += current.Value;
            current = current.Next;
        }

        return total;
    }

    public static int SolveRecursive(ListNode? head)
    {
        return head is null ? 0 : head.Value + SolveRecursive(head.Next);
    }
}
